ExternalModule.php

```diff
@@ -178,11 +181,14 @@ function setDefaultValues() {
                     }
 
                     // Getting previous event value.
-                    if (isset($data[$prev_event][$source_field])) {
-                        $default_value = $data[$prev_event][$source_field];
-                    } elseif (isset($data['repeat_instances'][$prev_event][""])) {
+                    // isset returns true for event instances when $prev_event_field_value is equal to an empty string ("").
+                    // An additional check to verify the value is not empty is required.
+                    $prev_event_field_value = $data[$prev_event][$source_field];
+                    if (isset($prev_event_field_value) && !empty($prev_event_field_value)) {
+                        $default_value = $prev_event_field_value;
+                    } elseif ($data['repeat_instances'][$prev_event][$source_form]) {
```

Comment:
This change is the source of the breakage for cross-event breakage, passing `$source_form` instead of `""` results in `null`

Reply:
I see, can you try the changes again with this setting enabled? 
![Screen Shot 2021-06-29 at 10 45 44 AM](https://user-images.githubusercontent.com/75623654/123818708-35f0d800-d8c7-11eb-9b45-1badaeb02b9b.png)

Reply:
That gets a little funky with imported data (timestamps for data entry are all the same when you do an upload), to really test it you've got to make a new record.

Reply:
I didn't realize I had that enabled when testing. When that setting is enabled both event instances and "Repeat entire event" are handled the same. 

I think the best option is to determine `x` in `$data[$prev_event][x]` on whether the event is an event instance or not. Do you know of a better way than `$_GET['instance'] > 1` to check if the current event is an event instance?

Thoughts?

Reply:
I think that's the easiest way to do it actually. There is a something (I'm blanking on the name, possibly part of the `$Proj` object) that tracks all repeating forms if the `$_GET` alone doesn't cut it.